sdk/resourcemanager/Azure.ResourceManager/src/ResourceIdentifier.cs

```diff
@@ -102,12 +102,20 @@ private void Init(ResourceIdentifier parent, ResourceType resourceType, string n
             _stringValue = parent == null ? RootStringValue : null;
         }
 
-        private static ResourceType ChooseResourceType(string resourceTypeName, ResourceIdentifier parent) => resourceTypeName.ToLowerInvariant() switch
+        private static ResourceType ChooseResourceType(string resourceTypeName, ResourceIdentifier parent)
         {
-            ResourceGroupsLowerKey => ResourceGroup.ResourceType,
-            SubscriptionsKey => Subscription.ResourceType,
-            _ => new ResourceType(parent.ResourceType, resourceTypeName)
-        };
```

Comment:
Is it possible to use a [when keyword](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression#case-guards) so we can keep the code succinct?